agent/recorder-agent.c

```diff
@@ -249,6 +249,106 @@ gboolean
   return TRUE;
 }
 
+gint
+find_chr (const gchar * str, gchar c)
+{
+  gint pos = -1;
+  gchar *p = NULL;
+  p = strchr (str, c);
+  if (p)
+    pos = p - str;
+  return pos;
+}
+
+gchar *
+check_filename (const gchar * fn, gint64 from, gint64 to)
+{
+  gint start, stop, len, pos;
+  g_autofree gchar *tmp = NULL;
+  gchar *time, *subtime;
+  gint64 time2 = 0;
+
+  start = 0;
+  stop = 0;
+  if (!g_str_has_prefix (fn, "hwangsae-recording-"))
+    return NULL;
+  start = 19;
+  if (g_str_has_suffix (fn, ".mp4"))
+    stop = 4;
+  if (g_str_has_suffix (fn, ".ts"))
+    stop = 3;
+  len = strlen (fn) - start - stop;
+  tmp = g_strndup (fn + start, len);
+
+  pos = find_chr (tmp, '-');
```

Comment:
For future reference have a look at `g_strsplit()` and `g_strsplit_set()` if you don't know them already. I think they could have helped you with parsing the filename in this function. It's probably not worth rewriting it though.

Reply:
It totally agree with you. I have already done a nicer implementation.